### DISCUSSION TITLE
Configure avoid tolls per country in route

### DISCUSSION THREAD

**Discussion by ADMUKHE** [valhalla/valhalla#4141] [General]
Is there any existing solution, where one can request route which avoids tolls specific to country. Or this would require changes in valhalla to have such specific route requests possible?

**Comment by kevinkreiser** [comment]
we have the data to be able to support such an option but at the moment the api doesnt allow any way of telling the library to do such things. we have avoid tolls, and we do know what country/state a part of the graph is in but we dont currently have a way to say, only care about avoiding tolls when in country X.

it would be somewhat interesting as a general feature to be able to tie admins to any of the hard/soft avoid-style costing options, or maybe we could allow different costing options per admin as a general feature.. however the interface/api to do such a thing would be very ugly (complex) in my opinion.
